Update to Angular 9

madoar opened the pull request:
<a href="https://gitpod.io/#https://github.com/madoar/angular-archwizard/pull/262"><img src="https://gitpod.io/api/apps/github/pbs/github.com/madoar/angular-archwizard.git/2e44d5689dd7af0a21b08681061f861dcc7f93d9.svg" /></a>

madoar commented on package.json:
```diff
@@ -74,19 +74,18 @@
   },
-  "peerDependencies": {
-    "@angular/common": ">= 5"
+  "peerDependencies": {},
```
@earshinov do you know if this is still valid?

earshinov replied:
Yes, I don't think that `@angular/common` is very useful in `peerDependencies` since angular-archwizard does not target any Angular versions specifically.

madoar replied:
I think the initial reason we added `@angular/common` to `peerDependencies` was to ensure that a compatible version of angular is used (i.e. a version greater than or equals to 5). If we remove this I'm not sure if we face any issues after the release.

earshinov replied:
Maybe you are right. Let's preserve it in `peerDependencies` to take care of users who might try to add the latest angular-archwizard to a project based on an old version of Angular

madoar replied:
Should we retain `5` or should we increase it to some more current version? I'm actually not sure up to which version Angular 9 is compatible with.

earshinov replied:
I would leave `5` and let users report if they encounter any problems. In theory Angular versions should be backward compatible (barring engine changes such as ViewEngine → Ivy)

madoar replied:
Done in 472e425

madoar commented on src/lib/components/wizard-completion-step.component.spec.ts:
```diff
@@ -24,7 +24,7 @@ import {WizardComponent} from './wizard.component';
 class WizardTestComponent {
 
-  @ViewChild(WizardComponent, {static: false})
+  @ViewChild(WizardComponent)
```
This is an automatic change by angular-cli

madoar commented on src/lib/components/wizard-navigation-bar.component.html:
```diff
@@ -14,7 +14,7 @@
       </div>
       <div class="step-indicator"
-        [ngStyle]="{ 'font-family': step.stepSymbolTemplate ? '' : step.navigationSymbol.fontFamily }">
+        [style]="{ 'font-family': step.stepSymbolTemplate ? '' : step.navigationSymbol.fontFamily }">
```
Angular 9 allows the use of the `style` and `class` bindings in addition to the "legacy" `ngStyle` and `ngClass` directives. See also section "**Improved CSS class and style binding**" in https://blog.angular.io/version-9-of-angular-now-available-project-ivy-has-arrived-23c97b63cfa3

madoar commented on src/lib/components/wizard-navigation-bar.component.spec.ts:
```diff
@@ -475,10 +469,7 @@ describe('WizardNavigationBarComponent', () => {
-      'horizontal': true, 'vertical': false, 'small': false,
-      'large-filled': true, 'large-filled-symbols': false, 'large-empty': false, 'large-empty-symbols': false
-    });
+    expect(navBarEl.classes).toEqual({ 'horizontal': true, 'large-filled': true });
```
It seems like the contract of `classes` has changed. The returned object now only contains the active classes, i.e. all classes that are true. This change affected 11 tests, which were "fixed" by me.

madoar commented on tsconfig.lib.prod.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "extends": "./tsconfig.lib.json",
+  "angularCompilerOptions": {
+    "enableIvy": false
```
This file has been generated by angular-cli. Most likely `enableIvy` should continue to be `false` because of [here](https://angular.io/guide/ivy#maintaining-library-compatibility). @earshinov what do you think? Should we change the value to `true`?

earshinov replied:
You are right about using `false`. It needs to be `false` to preserve backward compatibility of angular-archwizard library with applications still using ViewEngine.

earshinov commented:
Looks good to me. I think it can be merged (save the `peerDependencies` discussed [above](https://github.com/madoar/angular-archwizard/pull/262#discussion_r376709245))

madoar commented:
Ok I just tried to build the demo repository against this branch. Long story short it does not seem to work.

The initial problem was that typescript 3.7 contains some [breaking changes](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#class-field-mitigations) compared to typescript 3.5. This also answers the question whether an Angular 9 library is compatible with an Angular 8 (or earlier) application. The answer seems to be no.

After trying to update the demo repository to Typescript `3.7.5` and Angular 9 as well by using:

```bash
ng update @angular/core @angular/cli --allow-dirty
```

I got the following errors when calling `ng serve`:

```
Compiling @angular/core : es2015 as esm2015

Compiling @angular/common : es2015 as esm2015

Compiling @angular/platform-browser : es2015 as esm2015

Compiling @angular/platform-browser-dynamic : es2015 as esm2015

Compiling @angular/forms : es2015 as esm2015

Compiling @angular/router : es2015 as esm2015

chunk {main} main.js, main.js.map (main) 2.01 kB [initial] [rendered]
chunk {polyfills} polyfills.js, polyfills.js.map (polyfills) 684 bytes [initial] [rendered]
chunk {runtime} runtime.js, runtime.js.map (runtime) 6.15 kB [entry] [rendered]
chunk {scripts} scripts.js, scripts.js.map (scripts) 169 kB [entry] [rendered]
chunk {styles} styles.js, styles.js.map (styles) 1.37 MB [initial] [rendered]
chunk {vendor} vendor.js, vendor.js.map (vendor) 340 kB [initial] [rendered]
Date: 2020-02-14T22:14:28.481Z - Hash: 42da8404296f780fcc46 - Time: 16670ms

ERROR in src/app/completion-step-basic/completion-step-basic.component.html:25:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.

25   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/completion-step-basic/completion-step-basic.component.ts:5:16
    5   templateUrl: './completion-step-basic.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component CompletionStepBasicComponent.
src/app/completion-step-back-navigation/completion-step-back-navigation.component.html:25:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.

25   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'" awEnableBackLinks>
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/completion-step-back-navigation/completion-step-back-navigation.component.ts:5:16
    5   templateUrl: './completion-step-back-navigation.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component CompletionStepBackNavigationComponent.
src/app/optional-steps/optional-steps.component.html:38:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.

38   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/optional-steps/optional-steps.component.ts:5:16
    5   templateUrl: './optional-steps.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component OptionalStepsComponent.
src/app/default-navigation-mode/default-navigation-mode.component.html:37:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.

37   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/default-navigation-mode/default-navigation-mode.component.ts:5:16
    5   templateUrl: './default-navigation-mode.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component DefaultNavigationModeComponent.
src/app/configurable-navigation-mode/configurable-navigation-mode.component.html:57:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.

57   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/configurable-navigation-mode/configurable-navigation-mode.component.ts:5:16
    5   templateUrl: './configurable-navigation-mode.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component ConfigurableNavigationModeComponent.
src/app/free-navigation-mode/free-navigation-mode.component.html:50:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.

50   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/free-navigation-mode/free-navigation-mode.component.ts:5:16
    5   templateUrl: './free-navigation-mode.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component FreeNavigationModeComponent.
src/app/free-navigation-mode/free-navigation-mode.component.html:105:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.

105   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/free-navigation-mode/free-navigation-mode.component.ts:5:16
    5   templateUrl: './free-navigation-mode.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component FreeNavigationModeComponent.
src/app/default-wizard-step/default-wizard-step.component.html:37:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.

37   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/default-wizard-step/default-wizard-step.component.ts:5:16
    5   templateUrl: './default-wizard-step.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component DefaultWizardStepComponent.
src/app/initially-completed-wizard-steps/initially-completed-wizard-steps.component.html:42:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.

42   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  src/app/initially-completed-wizard-steps/initially-completed-wizard-steps.component.ts:5:16
    5   templateUrl: './initially-completed-wizard-steps.component.html',
                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    Error occurs in the template of component InitiallyCompletedWizardStepsComponent.

** Angular Live Development Server is listening on localhost:4200, open your browser on http://localhost:4200/ **

Date: 2020-02-14T22:14:33.573Z - Hash: 880413209ce45e690cf1
3 unchanged chunks
chunk {main} main.js, main.js.map (main) 726 kB [initial] [rendered]
chunk {polyfills} polyfills.js, polyfills.js.map (polyfills) 198 kB [initial] [rendered]
chunk {vendor} vendor.js, vendor.js.map (vendor) 3.38 MB [initial] [rendered]
Time: 4763ms
: Compiled successfully.
    
    ERROR in src/app/completion-step-basic/completion-step-basic.component.html:25:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
    1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
    2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
    3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.
    
    25   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    
      src/app/completion-step-basic/completion-step-basic.component.ts:5:16
        5   templateUrl: './completion-step-basic.component.html',
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        Error occurs in the template of component CompletionStepBasicComponent.
    src/app/completion-step-back-navigation/completion-step-back-navigation.component.html:25:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
    1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
    2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
    3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.
    
    25   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'" awEnableBackLinks>
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    
      src/app/completion-step-back-navigation/completion-step-back-navigation.component.ts:5:16
        5   templateUrl: './completion-step-back-navigation.component.html',
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        Error occurs in the template of component CompletionStepBackNavigationComponent.
    src/app/optional-steps/optional-steps.component.html:38:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
    1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
    2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
    3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.
    
    38   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    
      src/app/optional-steps/optional-steps.component.ts:5:16
        5   templateUrl: './optional-steps.component.html',
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        Error occurs in the template of component OptionalStepsComponent.
    src/app/default-navigation-mode/default-navigation-mode.component.html:37:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
    1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
    2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
    3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.
    
    37   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    
      src/app/default-navigation-mode/default-navigation-mode.component.ts:5:16
        5   templateUrl: './default-navigation-mode.component.html',
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        Error occurs in the template of component DefaultNavigationModeComponent.
    src/app/configurable-navigation-mode/configurable-navigation-mode.component.html:57:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
    1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
    2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
    3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.
    
    57   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    
      src/app/configurable-navigation-mode/configurable-navigation-mode.component.ts:5:16
        5   templateUrl: './configurable-navigation-mode.component.html',
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        Error occurs in the template of component ConfigurableNavigationModeComponent.
    src/app/free-navigation-mode/free-navigation-mode.component.html:50:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
    1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
    2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
    3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.
    
    50   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    
      src/app/free-navigation-mode/free-navigation-mode.component.ts:5:16
        5   templateUrl: './free-navigation-mode.component.html',
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        Error occurs in the template of component FreeNavigationModeComponent.
    src/app/free-navigation-mode/free-navigation-mode.component.html:105:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
    1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
    2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
    3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.
    
    105   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    
      src/app/free-navigation-mode/free-navigation-mode.component.ts:5:16
        5   templateUrl: './free-navigation-mode.component.html',
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        Error occurs in the template of component FreeNavigationModeComponent.
    src/app/default-wizard-step/default-wizard-step.component.html:37:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
    1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
    2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
    3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.
    
    37   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    
      src/app/default-wizard-step/default-wizard-step.component.ts:5:16
        5   templateUrl: './default-wizard-step.component.html',
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        Error occurs in the template of component DefaultWizardStepComponent.
    src/app/initially-completed-wizard-steps/initially-completed-wizard-steps.component.html:42:30 - error NG8002: Can't bind to 'stepTitle' since it isn't a known property of 'aw-wizard-completion-step'.
    1. If 'aw-wizard-completion-step' is an Angular component and it has 'stepTitle' input, then verify that it is part of this module.
    2. If 'aw-wizard-completion-step' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.
    3. To allow any property add 'NO_ERRORS_SCHEMA' to the '@NgModule.schemas' of this component.
    
    42   <aw-wizard-completion-step [stepTitle]="'Completion steptitle'">
                                    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    
      src/app/initially-completed-wizard-steps/initially-completed-wizard-steps.component.ts:5:16
        5   templateUrl: './initially-completed-wizard-steps.component.html',
                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
        Error occurs in the template of component InitiallyCompletedWizardStepsComponent.
```

After a bit of additional search I found out that these errors could be prevented by [deactivating Ivy and aot](https://next.angular.io/guide/ivy#opting-out-of-ivy-in-version-9).

After deactivating Ivy and aot `ng serve` succeeds but my browser shows a white window with the following error in the debug terminal:

```
Unhandled Promise rejection: Failed to load wizard.component.html ; Zone: <root> ; Task: Promise.then ; Value: Failed to load wizard.component.html undefined
```

I'm a bit clueless how to continue. @earshinov do you by chance have an idea?

earshinov commented:
Hmm, there shouldn't be so much trouble. I will try to take a look tomorrow.

earshinov commented:
@madoar , you might have noticed that the demo project does not recognize the `stepTitle` input of the `WizardCompletionStepComponent`. To resolve the problem you just has to add `@Directive()` decorator to the `WizardCompletionStep` abstract class.

Angular migration script applies it automatically to the base `WizardStep` classs, but misses `WizardCompletionStep`. I guess it is an oversight of the Angular team and they would appreciate a bug report (if there isn't one already).

I have pushed changes I had to make to
- https://github.com/earshinov/angular-archwizard/commits/angular-9-update
- https://github.com/earshinov/angular-archwizard-demo/commits/angular-9-update

Also I would recommend to always build the library with `enableIvy: false`, i.e. remove tsconfig.lib.prod.js and write `"angularCompilerOptions": { "enableIvy": false }` into the root tsconfig.json.

madoar commented:
@earshinov thank you for investigating this! I've applied your commits to my branches.

> Also I would recommend to always build the library with enableIvy: false, i.e. remove tsconfig.lib.prod.js and write "angularCompilerOptions": { "enableIvy": false } into the root tsconfig.json.

I'm not sure which other configuration files I need to modify, and how to modify them, to make this change work. Can you by chance do the necessary changes and I apply them afterwards on my branch?

earshinov commented:
Sure. Please see this commit: https://github.com/earshinov/angular-archwizard/commit/e7675427d43ced7b380a7264e33c3512f93f7e15

madoar commented:
> Sure. Please see this commit: earshinov@e767542

@earshinov thanks again for your help!

I've just pushed some additional changes/improvements. More specific I have:
- updated the dependencies again using `ng update`
- updated the `README` and `package.json` files with the new minimum supported Angular version 9

@earshinov could you please take another look at this PR and see if I have missed anything? If everything is all right I think we can merge this.

madoar commented:
I forgot to mention that it would also be nice if we could update the remaining packages that are kind of independent of Angular as well, e.g. `scss-bundle`. Sadly I couldn't get it working with `scss-bundle` version 3.

earshinov commented:
The changes look good to me. As for scss-bundle, it can be upgraded at a later date.
